src/fieri/app/models/publish_worker.rb

```diff
@@ -0,0 +1,58 @@
+require 'sidekiq'
+require 'net/http'
+
+class PublishWorker
+  include ::Sidekiq::Worker
+
+  def perform(cookbook_name)
+    parsed = JSON.parse(get_supermarket_response(cookbook_name))
+
+    failure = false
+    publish_feedback = ''
+
+    unless cookbook_exists?(parsed, cookbook_name)
+      failure = true
+      publish_feedback += "#{cookbook_name} not found in Supermarket\n"
+    end
+
+    unless cookbook_not_deprecated?(parsed)
+      failure = true
+      publish_feedback += "#{cookbook_name} is deprecated\n"
+    end
+
+    unless cookbook_not_for_adoption?(parsed)
```

Comment:
After staring at the private methods for a bit, I've come around to liking all the dependency injection going on in here. ✨ 💉